src/java/main/org/apache/zookeeper/server/persistence/FileTxnSnapLog.java

```diff
@@ -167,6 +175,16 @@ public long restore(DataTree dt, Map<Long, Integer> sessions,
             PlayBackListener listener) throws IOException {
         long deserializeResult = snapLog.deserialize(dt, sessions);
         FileTxnLog txnLog = new FileTxnLog(dataDir);
+        boolean suspectEmptyDB;
+        File initFile = new File(dataDir.getParent(), "initialize");
+        if (initFile.exists()) {
+            if (!initFile.delete()) {
+                throw new IOException("Unable to delete initialization file " + initFile.toString());
+            }
+            suspectEmptyDB = false;
+        } else {
+            suspectEmptyDB = !autoCreateDB;
```

Comment:
I tempted to do put the log line on the other side of the conditional since this side is the expected case. We should only delete an initialize file once in the lifecycle of a given server while the check against `autoCreateDB` will happen every other time the server is restarted.